### PR TITLE
Save rendered HTML in post content

### LINKED ISSUE #26: Store render in the content body
Storing the content in the post's body offers some advantages (search, possibly performance, clearer diffs).

Generate render on or near the `wp_insert_post_data` filter.

As a followup and alternative to #15.

### PR DESCRIPTION
**Pull request by roborourke**
Fixes #26

Alternative to #15

cc @mattheu @peterwilsoncc

### CODE REVIEW

File: inc/class-builder-post-meta.php [hunk 1]
```diff
@@ -75,32 +76,19 @@ public function output( $post ) {
...
-			if ( json_last_error() ) {
-				$data = json_decode( stripslashes( $json ) );
-			}
+	public function wp_insert_post_data( $post_data ) {
 
-			if ( ! json_last_error()  && $nonce && wp_verify_nonce( $nonce, $this->id ) ) {
-				$this->save_data( $post_id, $data );
-			}
+		if ( $data = $this->get_post_data() ) {
+			$post_data['post_content'] = $this->get_rendered_data( $data );
```
Comment (peterwilsoncc) [inline inc/class-builder-post-meta.php:93]: Data passed to/expected following `wp_insert_post_data` is:
- `sanitize_post_field` has run with the `db` context
- slashed 
- the `post_content` columns character set has been tested  and `wp_encode_emoji` run if needs be.

Some of these will be more or less relevant for our purposes here.

Comment (roborourke) [inline inc/class-builder-post-meta.php:93]: Cheers Peter, will check the `wp_insert_post()` function. Thought that processing happened afterwards!

**Comment by roborourke** [comment]
@peterwilsoncc that should be it now. Thanks again, every day is a school day!

**Comment by mattheu** [comment]
Thanks Rob. I'll give this a test. Maybe this evening, otherwise monday.

**Comment by mattheu** [comment]
This is looking good.

However I've run into a slight problem. `wp_insert_post_data` assumes the post ID is set, but this is not the case for a revision, and I'm getting errors.

I think its probably OK to just check the post ID is set before getting the data. But wanted to run it by you in case you thought we needed to do anything special for revisions.

**Comment by roborourke** [comment]
@mattheu d'oh, cheers for testing! So no `$postarr` param? I'll trace through with some breakpoints. We want the post content to be set in the revisions so as long as that bit is working it should be ok.

**Comment by mattheu** [comment]
So no ID when creating the revision. The post content gets copied across, which I guess is correct.

**Comment by roborourke** [comment]
@mattheu that should be it now, seems to work well and no errors that I've seen in the error log on the edit screen or revisions UI after the last commit

**Comment by mattheu** [comment]
👍 Thanks!

**Pull request merged**